pkg/apis/pipeline/v1/pipeline_defaults.go

```diff
@@ -50,12 +50,12 @@ func (ps *PipelineSpec) SetDefaults(ctx context.Context) {
 func (pt *PipelineTask) SetDefaults(ctx context.Context) {
 	cfg := config.FromContextOrDefaults(ctx)
 	if pt.TaskRef != nil {
-		if pt.TaskRef.Kind == "" {
-			pt.TaskRef.Kind = NamespacedTaskKind
-		}
 		if pt.TaskRef.Name == "" && pt.TaskRef.Resolver == "" {
 			pt.TaskRef.Resolver = ResolverName(cfg.Defaults.DefaultResolverType)
 		}
+		if pt.TaskRef.Kind == "" && pt.TaskRef.Resolver == "" {
+			pt.TaskRef.Kind = NamespacedTaskKind
+		}
 	}
```

Comment:
This will continue to work even after [TEP-0154](https://github.com/tektoncd/community/blob/main/teps/0154-concise-remote-resolver-syntax.md) is done